enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,644 @@
+1. The Agent is marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. The Baremetal Agent Controller adds inspection data found on the Agent's Status to the BareMetalHost.
+1. When the agent is in a ready state, installation of that host begins.
+
+#### Future Use of MachinePools and Machine API
+
+This section discusses potential future use of hive's MachinePool and the
+Machine API as a preview of how the proposed agent-based installation can later
+fit a Machine-driven provisioning workflow.
+
+Hive has a MachinePool resource type that exists on a hub cluster and offers a
+way to manage MachineSets on spoke clusters. This proposal does not utilize
+any part of the Machine API because it does not (yet) add agent-based
+provisioning support to Machine API providers.
+
+A separate effort is underway to implement Centralized Machine Management,
+where Machine API providers and related resources would all be implemented on
+the hub cluster instead of the spoke cluster. Rather than use a MachinePool on
+the hub as a proxy for a remote MachineSet resource, the entire Machine API
+affecting a spoke cluster would be available for use on the Hub. That will be
+the starting point for integrating agent-based installation with Machines.
+
+Typically with the Machine API, a MachineSet gets scaled up, one or more
+Machines get created, and then a controller (part of a particular platform's
+machine-api-provider) interacts with a provider API (such as a public cloud
+API) to create or obtain a host per Machine that can be provisioned as Nodes in
+the cluster.
+
+Integrating agent-based provisioning into that flow will involve matching each
+Machine resource with an Agent resource in a similar way to how the bare metal
+provider currently matches a Machine with an availbale BareMetalHost. On a bare
+metal platform, Agents will likely continue to be created in advance,
+independently of a Machine. It is possible that other platforms may create
+Agents on-demand when a Machine appears.
+
+#### Hive Cluster Adoption
+
+Hive is able to adopt existing clusters that it did not create. This is done by
+creating a ClusterDeployment where `spec.installed == true` and providing a
+kubeconfig.
+
+In order to adopt a cluster and then add workers using an agent-based approach,
+the user must additionally:
+* specify the agent label selector on the ClusterDeployment's Spec
+* add a reference to the ClusterDeployment on one or more InstallEnvs
+
+Otherwise there are no differences in the InstallEnv or the workflow as
+described above.
+
+### Risks and Mitigations
+
+#### API Versions and Potential Change
```

Comment:
Since I know y'all are considering revisiting the plan in this section, and since it's a pretty important detail, I'll leave the hold in place for now until that discussion comes to a resolution.